File: 3737-count-subarrays-with-majority-element-i/3737-count-subarrays-with-majority-element-i.go
func countMajoritySubarrays(nums []int, target int) int {
    ans := 0
    for i := 0; i < len(nums); i++ {
        count := 0
        for j := i; j < len(nums); j++ {
            if nums[j] == target {
                count++
            }

            if count > (j - i + 1) / 2 {
                ans++
            }
        }
    }

    return ans
}

// array, brute force
// time: O(n^2)
// space: O(1)

// try to get time O(n) (using prefix sum cmiiw)

File: 3737-count-subarrays-with-majority-element-i/3737-count-subarrays-with-majority-element-i.py
class Solution:
    def countMajoritySubarrays(self, nums: List[int], target: int) -> int:
        ans = 0
        for i in range(len(nums)):
            count = 0
            for j in range(i, len(nums), 1):
                if nums[j] == target:
                    count += 1
                
                if count > (j - i + 1) // 2:
                    ans += 1
        
        return ans

# array, brute force
# time: O(n^2)
# space: O(1)